vidarr-server/src/main/java/ca/on/oicr/gsi/vidarr/server/PriorityByWorkflow.java

```diff
@@ -117,29 +97,38 @@ public synchronized ConsumableResourceResponse request(
               + "values should be one of the following: %s", workflowName, workflowPriority,
               acceptedPriorities.stream().map(String::valueOf).collect(Collectors.joining(", "))));
     }
+
+
     final var state = workflows.get(workflowName);
-    if (state == null) {
-      return ConsumableResourceResponse.error(
-          String.format("Internal Vidarr error: The %s workflow run priority has not been configured properly.", workflowName));
+    if (state == null || state.waiting.isEmpty()) {
+      return ConsumableResourceResponse.AVAILABLE;
+    }
+
+    // If this workflow has already been seen
+    // Add the current run to the waitlist
+    // ensuring it replaces previous runs with the same ID, accounting for if the priority has changed
+    set(workflowName, vidarrId, input);
+    SimpleEntry resourcePair = new SimpleEntry(vidarrId, workflowPriority);
```

Comment:
Why create this Entry (which, btw, raw use of parameterized class warning) when you've already set the waiting state with the same information? And even if you haven't, much cleaner to have a remove implementation in WaitingState that uses getByKey the way add does, since your add implementation ensures there can't be two entries for the same vidarr id, so you don't need to know priority to remove it